### ISSUE TITLE
[SF_ACC][List]: [UI5 WebComponents List - Keyboard Accessibility Issue with Delete Selection Mode]

### ISSUE THREAD

**Issue by cptDev6** [UI5/webcomponents#13220]
### Bug Description

The UI5 WebComponents List component with `selectionMode="Delete"` is not properly accessible via keyboard navigation. Users cannot tab to focus the Delete/cross-icon from the list to remove items, making this feature unusable for keyboard-only users.

### Expected Behaviour

- Users should be able to navigate to delete/cross icon of the list items using Tab key

### Isolated Example

https://ui5.github.io/webcomponents-react/v2/?path=/docs/data-display-list--docs

### Steps to Reproduce

1. In ui5 web components, open List component- https://ui5.github.io/webcomponents-react/v2/?path=/docs/data-display-list--docs
2. Select selection mode as delete
3. And see the cross icon is not jeyboard accessible
...

https://github.com/user-attachments/assets/6f4d8f12-dd3d-454e-a172-c9449a043a72

### Priority

High

### UI5 Web Components Version

2.19.1

### Browser

Chrome

### Operating System

Windows

### Organization

SAP Labs

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by kskondov** [comment]
Hello @UI5/ui5-team-ctr-pir the delete button in the list when in selectionMode delete is not accessible via keyboard. Can you please take a look at this issues?
